File: src/main/java/MoveZeroes/L07152055/MoveZeroes.java
public class MoveZeroes {
    public static void movezeroes(int[] nums) {
        for(int i = 0, j = 0; i < nums.length; i++) {
            if(nums[i] != 0) {
                int tmp = nums[i];
                nums[i] = nums[j];
                nums[j++] = tmp;
            }
        }
    }
}

File: src/main/java/MoveZeroes/GaryLinTouchWorld/MoveZeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    let length = nums.length
    for(let i = 0 ; i < length ; i++){
        if(nums[i] === 0){
            nums.splice(i,1);
            nums.push(0);
            i = -1;
            length -= 1;
        }
    }
    return nums;
};